numpyro/infer/util.py

```diff
@@ -286,12 +331,16 @@ def body_fn(state):
     if prototype_params is not None:
         init_state = (0, rng, prototype_params, False)
     else:
-        init_state = body_fn((0, rng, None, None))
+        _, _, prototype_params, is_valid = init_state = body_fn((0, rng, None, None))
+        if not_jax_tracer(is_valid):
```

Comment:
Just for my understanding, could you explain why we need this?

Reply:
This is not necessary for the usual case of SVI. I just want to make sure that if someone runs `svi` inside vmap/jit (e.g. vmap over different learning rate/initial seed/shards of data), the check `if is_valid` will trigger an error.